### ISSUE TITLE
tuned model not being loaded or reflected in logs (DIA-NN 2.1)

### ISSUE THREAD

**Issue by palette221999** [vdemichev/DiaNN#1922]
Hi DIA-NN team,

I am attempting to use a fine-tuned retention time model for my DIA-NN search. Although I have specified the --rt-model and --tokens parameters in the command line, the execution log appears identical to a standard search without these parameters. There is no indication in the log that the .pt model is being successfully loaded or utilized.

**1. use tuned model**
diann.exe --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_A1.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_A2.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_A3.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_B1.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_B2.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_B3.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_C1.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_C2.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_C3.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_D1.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_D2.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_D3.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_E1.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_E2.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_E3.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_F1.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_F2.raw
" --f "E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_F3.raw
" --lib "E:\users3\yppeng\di2\lib\report-lib.parquet" --threads 36 --verbose 1 --out "E:\users3\yppeng\di2\diann_result_all_rt\report.parquet" --qvalue 0.01 --matrices  --out-lib "E:\users3\yppeng\di2\diann_result_all_rt\report-lib.parquet" --gen-spec-lib --unimod4 --var-mods 1 --var-mod UniMod:35,15.994915,M --peptidoforms --rt-profiling --fixed-mod Dimethyl, 28.0313, nK
--original-mods
--mod Dimethyl, 28.0313
--tokens E:\users3\yppeng\di2\lib\tune_lib.dict.txt
--rt-model E:\users3\yppeng\di2\lib\tune_lib.tuned_rt.pt

DIA-NN 2.1.0 Academia  (Data-Independent Acquisition by Neural Networks)
Compiled on Mar 25 2025 10:59:54
Current date and time: Wed Apr  8 22:53:30 2026
CPU: GenuineIntel Intel(R) Xeon(R) Gold 6254 CPU @ 3.10GHz
SIMD instructions: AVX AVX2 AVX512CD AVX512F FMA SSE4.1 SSE4.2 
Logical CPU cores: 36
Thread number set to 36
Output will be filtered at 0.01 FDR
Precursor/protein x samples expression level matrices will be saved along with the main report
A spectral library will be generated
Cysteine carbamidomethylation enabled as a fixed modification
Maximum number of variable modifications set to 1
Modification UniMod:35 with mass delta 15.9949 at M will be considered as variable
Peptidoform scoring enabled
The spectral library (if generated) will retain the original spectra but will include empirically-aligned RTs
Modification Dimethyl with mass delta 28.0313 at  nK will be considered as fixed
DIA-NN will not attempt to convert library modifications to the UniMod format
Modification Dimethyl with mass delta 28.0313 added to the list of recognised modifications for spectral library-based search
DIA-NN will automatically optimise the mass accuracy for the first run of the experiment, use this mode for preliminary analyses only
WARNING: protein inference is enabled but no FASTA provided - is this intended?
The following variable modifications will be localised: UniMod:35 

18 files will be processed
[0:00] Loading spectral library E:\users3\yppeng\di2\lib\report-lib.parquet
[5:15] Spectral library loaded: 26619 protein isoforms, 36721 protein groups and 7218977 precursors in 2217715 elution groups.
[5:44] Initialising library
[6:58] Saving the library to E:\users3\yppeng\di2\lib\report-lib.parquet.skyline.speclib
WARNING: it is strongly recommended to enable MBR when analysing with a large library, if this is a quantitative analysis

[8:18] File #1/18
[8:18] Loading run E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_A1.raw
[10:01] Pre-processing...
[10:24] 1092 MS1 and 218373 MS2 scans in 1092 (inferred) and 1092 (encoded) cycles, 4414144 precursors in range
[10:41] Calibrating with mass accuracies 21 (MS1), 25 (MS2)
[13:08] RT window set to 1.96025
[13:08] Peak width: 2.984
[13:08] Scan window radius set to 6
[13:08] Recommended MS1 mass accuracy setting: 3.3 ppm
[16:51] Optimised mass accuracy: 7 ppm
[17:59] Searching decoys
[20:05] Main search
[23:51] Removing low confidence identifications
[24:18] Removing interfering precursors
[24:53] Training neural networks on 102094 target and 64750 decoy PSMs
[27:37] Training neural networks on 102094 target and 62129 decoy PSMs
[29:15] Number of IDs at 0.01 FDR: 48853
[29:16] Precursors at 1% peptidoform FDR: 44647
[29:18] Calculating protein q-values
WARNING: 739 precursors were wrongly annotated in the library as proteotypic
[29:19] Number of protein isoforms identified at 1% FDR: 7417 (precursor-level), 6854 (protein-level) (inference performed using proteotypic peptides only)
[29:19] Quantification
[29:21] Precursors with scored PTMs at 1% FDR: 3324 out of 3984 considered
[29:21] Precursors with all scored PTM sites unoccupied at 1% FDR: 41323
[29:21] Precursors with PTMs localised (when required) with > 90% confidence: 3181 out of 3324
[29:24] Quantification information saved to E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_A1.raw.quant

**2. Not used**
diann.exe --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_A1.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_A2.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_A3.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_B1.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_B2.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_B3.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_C1.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_C2.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_C3.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_D1.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_D2.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_D3.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_E1.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_E2.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_E3.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_F1.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_F2.raw  --f E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_F3.raw  --lib E:\users3\yppeng\di1\lib1\report-lib.parquet --threads 36 --verbose 1 --out E:\users3\yppeng\di1\diannn_result_all\report.parquet --qvalue 0.01 --matrices --out-lib E:\users3\yppeng\di1\diannn_result_all\report-lib.parquet --gen-spec-lib --unimod4 --var-mods 1 --var-mod UniMod:35,15.994915,M --peptidoforms --reanalyse --rt-profiling --fixed-mod Dimethyl, 28.0313, nK  --original-mods 

Thread number set to 36
Output will be filtered at 0.01 FDR
Precursor/protein x samples expression level matrices will be saved along with the main report
A spectral library will be generated
Cysteine carbamidomethylation enabled as a fixed modification
Maximum number of variable modifications set to 1
Modification UniMod:35 with mass delta 15.9949 at M will be considered as variable
Peptidoform scoring enabled
MBR enabled; .quant files will only be saved to disk during the first pass
The spectral library (if generated) will retain the original spectra but will include empirically-aligned RTs
Modification Dimethyl with mass delta 28.0313 at  nK will be considered as fixed
DIA-NN will not attempt to convert library modifications to the UniMod format
DIA-NN will automatically optimise the mass accuracy for the first run of the experiment, use this mode for preliminary analyses only
WARNING: protein inference is enabled but no FASTA provided - is this intended?
The following variable modifications will be localised: UniMod:35 

18 files will be processed
[0:00] Loading spectral library E:\users3\yppeng\di1\lib1\report-lib.parquet
[2:20] Spectral library loaded: 26619 protein isoforms, 36721 protein groups and 7218977 precursors in 2217715 elution groups.
[2:31] Initialising library
[3:24] Saving the library to E:\users3\yppeng\di1\lib1\report-lib.parquet.skyline.speclib

First pass: generating a spectral library from DIA data

[3:36] File #1/18
[3:36] Loading run E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_A1.raw
[5:03] Pre-processing...
[5:14] 1092 MS1 and 218373 MS2 scans in 1092 (inferred) and 1092 (encoded) cycles, 4414144 precursors in range
[5:22] Calibrating with mass accuracies 21 (MS1), 25 (MS2)
[7:30] RT window set to 1.96025
[7:30] Peak width: 2.984
[7:30] Scan window radius set to 6
[7:30] Recommended MS1 mass accuracy setting: 3.3 ppm
[10:13] Optimised mass accuracy: 7 ppm
[11:16] Searching decoys
[12:38] Main search
[15:00] Removing low confidence identifications
[15:19] Removing interfering precursors
[15:32] Training neural networks on 102094 target and 64750 decoy PSMs
[16:45] Training neural networks on 102094 target and 62129 decoy PSMs
[18:01] Number of IDs at 0.01 FDR: 48853
[18:02] Precursors at 1% peptidoform FDR: 44647
[18:04] Calculating protein q-values
WARNING: 739 precursors were wrongly annotated in the library as proteotypic
[18:04] Number of protein isoforms identified at 1% FDR: 7417 (precursor-level), 6854 (protein-level) (inference performed using proteotypic peptides only)
[18:04] Quantification
[18:06] Precursors with scored PTMs at 1% FDR: 3324 out of 3984 considered
[18:06] Precursors with all scored PTM sites unoccupied at 1% FDR: 41323
[18:06] Precursors with PTMs localised (when required) with > 90% confidence: 3181 out of 3324
[18:08] Quantification information saved to E:\users3\yppeng\di1\data\20240408_OA1_Evo12_31min_TiHe_SA_H032_E32_F-40_A1.raw.quant

**Comment by vdemichev** [comment]
Hi,

The models will only be used when a prediction of a spectral library (e.g. from FASTA digest) is performed, i.e. they do not affect raw data search.

Btw, we are preparing a new release that will feature an interactive Wizard that guides one through configuring DIA-NN, including for fine-tuning.

Best,
Vadim

**Comment by palette221999** [comment]
Thanks! I've got it now
